pkg/apis/apps/v1alpha1/helm.go

```diff
@@ -85,6 +85,11 @@ type HelmOptions struct {
 	// +kubebuilder:validation:Pattern=`^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&\(\)\*\+,;=.]+$`
 	Repository string `json:"repo"`
 
+	// AuthSecret  is the name of the secret that contains the auth information for the chart repository.
+	//
+	// +optional
+	AuthSecret AuthSecret `json:"authSecret"`
```

Comment:
s/AuthSecret/ChartPullSecret

WDYT?

Reply:
Moreover, since this field is optional, `omitempty` should be added to json tag as well.